lib/rules/id-length.js

```diff
@@ -0,0 +1,108 @@
+        obj[item] = true;
+
+        return obj;
+    }, {});
+
+    var SUPPORTED_EXPRESSIONS = {
+        "AssignmentExpression": function (parent, node) {
+            return parent.left.type === "MemberExpression" &&
+                !parent.left.computed && parent.left.property === node;
+        },
+        "VariableDeclarator": function (parent, node) {
+            return parent.id === node;
+        },
+        "ObjectExpression": function (parent, node) {
+            return node.parent.key === node;
+        },
+        "FunctionExpression": true,
+        "FunctionDeclaration": true,
+        "CatchClause": true
+    };
+
+    // Disable excluded expressions
+    (Array.isArray(options.disableFor) ? options.disableFor : []).forEach(function (expression) {
+        SUPPORTED_EXPRESSIONS[expression] = false;
+    });
+
+    return {
+        Identifier: function (node) {
+            // Leading and trailing `_` and `$` are commonly used to flag
+            // private/protected identifiers or jQuery instances, strip them
+            var name = node.name.replace(/^[$_]{0,2}(.+)[$_]{0,2}$/g, "$1");
+            var effectiveParent = (node.parent.type === "MemberExpression" || node.parent.type === "Property") ?
+                                  node.parent.parent : node.parent;
+
+            var isShort = name.length < minLength;
+            var isLong = name.length > maxLength;
+            if (!(isShort || isLong) || exceptions[name]) {
+                return;  // Nothing to report
+            }
+
+            var isValidExpression = SUPPORTED_EXPRESSIONS[effectiveParent.type];
+
+            if (isValidExpression && (isValidExpression === true || isValidExpression(effectiveParent, node))) {
+                context.report(
+                    node,
+                    isShort ?
+                        "Identifier name '{{name}}' is too short. (< {{min}})" :
+                        "Identifier name '{{name}}' is too long. (> {{max}})",
+                    { name: node.name, min: minLength, max: maxLength }
+                );
+            }
+        }
+    };
+};
+
+module.exports.schema = [
+    {
+        "type": "object",
+        "properties": {
+            "min": {
+                "type": "number"
+            },
+            "max": {
+                "type": "number"
+            },
+            "exceptions": {
+                "type": "array",
+                "uniqueItems": true,
+                "items": {
+                    "type": "string"
+                }
+            },
+            "disableFor": {
+                "type": "array",
+                "uniqueItems": true,
+                "items": {
+                    "enum": [
+                        "AssignmentExpression",
+                        "VariableDeclarator",
+                        "ObjectExpression",
+                        "FunctionExpression",
+                        "FunctionDeclaration",
+                        "CatchClause"
```

Comment:
Please mention this in docs also that these are the available options for `disableFor`. Makes it easier for people.